src/cider/type/plan/FunctionExpr.cpp

```diff
@@ -0,0 +1,307 @@
+      return JITTypeTag::INT16;
+    case ExtArgumentType::Int32:
+      return JITTypeTag::INT32;
+    case ExtArgumentType::Int64:
+      return JITTypeTag::INT64;
+    case ExtArgumentType::Float:
+      return JITTypeTag::FLOAT;
+    case ExtArgumentType::Double:
+      return JITTypeTag::DOUBLE;
+    case ExtArgumentType::ArrayInt64:
+    case ExtArgumentType::ArrayInt32:
+    case ExtArgumentType::ArrayInt16:
+    case ExtArgumentType::ArrayBool:
+    case ExtArgumentType::ArrayInt8:
+    case ExtArgumentType::ArrayDouble:
+    case ExtArgumentType::ArrayFloat:
+    case ExtArgumentType::ColumnInt64:
+    case ExtArgumentType::ColumnInt32:
+    case ExtArgumentType::ColumnInt16:
+    case ExtArgumentType::ColumnBool:
+    case ExtArgumentType::ColumnInt8:
+    case ExtArgumentType::ColumnDouble:
+    case ExtArgumentType::ColumnFloat:
+    case ExtArgumentType::TextEncodingNone:
+    case ExtArgumentType::ColumnListInt64:
+    case ExtArgumentType::ColumnListInt32:
+    case ExtArgumentType::ColumnListInt16:
+    case ExtArgumentType::ColumnListBool:
+    case ExtArgumentType::ColumnListInt8:
+    case ExtArgumentType::ColumnListDouble:
+    case ExtArgumentType::ColumnListFloat:
+      return JITTypeTag::VOID;
+    default:
+      CHECK(false);
+  }
+  CHECK(false);
+  return JITTypeTag::INVALID;
+}
+
+bool ext_func_call_requires_nullcheck(const Analyzer::FunctionOper* function_oper) {
+  const auto& func_ti = function_oper->get_type_info();
+  for (size_t i = 0; i < function_oper->getArity(); ++i) {
+    const auto arg = function_oper->getArg(i);
+    const auto& arg_ti = arg->get_type_info();
+    if ((func_ti.is_array() && arg_ti.is_array()) ||
+        (func_ti.is_bytes() && arg_ti.is_bytes())) {
+      // If the function returns an array and any of the arguments are arrays, allow NULL
+      // scalars.
+      // TODO: Make this a property of the FunctionOper following `RETURN NULL ON NULL`
+      // semantics.
+      return false;
+    } else if (!arg_ti.get_notnull() && !arg_ti.is_buffer()) {
+      return true;
+    } else {
+      continue;
+    }
+  }
+  return false;
+}
+
+inline SQLTypeInfo get_sql_type_from_JIT_type_tag(const JITTypeTag jit_type) {
+  switch (jit_type) {
+    case JITTypeTag::FLOAT:
+      return SQLTypeInfo(kFLOAT, false);
+    case JITTypeTag::DOUBLE:
+      return SQLTypeInfo(kDOUBLE, false);
+    case JITTypeTag::BOOL:
+      return SQLTypeInfo(kBOOLEAN, false);
+    case JITTypeTag::INT8:
+      return SQLTypeInfo(kTINYINT, false);
+    case JITTypeTag::INT16:
+      return SQLTypeInfo(kSMALLINT, false);
+    case JITTypeTag::INT32:
+      return SQLTypeInfo(kINT, false);
+    case JITTypeTag::INT64:
+      return SQLTypeInfo(kBIGINT, false);
+    default:
+      LOG(FATAL) << "Unrecognized jit type for SQL type";
+  }
+  UNREACHABLE();
+  return SQLTypeInfo();
+}
+
+// Generates code which returns true if at least one of the arguments is NULL.
+JITValue* codegenFunctionOperNullArgForArrow(
```

Comment:
Seems the `null` can be determined by the arg type info directly instead of check on the JITValues. You can leverage existing method `ext_func_call_requires_nullcheck(const Analyzer::FunctionOper* function_oper) ` by making a copy in this file.